akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerDemuxSpec.scala

```diff
@@ -0,0 +1,57 @@
+package akka.http.impl.engine.http2
+
+import akka.http.impl.engine.http2.FrameEvent.{ ParsedHeadersFrame, Setting, SettingsAckFrame, SettingsFrame, WindowUpdateFrame }
+import akka.http.impl.engine.http2.Http2Protocol.SettingIdentifier
+import akka.http.impl.util.AkkaSpecWithMaterializer
+import akka.http.scaladsl.settings.Http2ServerSettings
+import akka.stream.scaladsl.{ BidiFlow, Flow, Keep }
+import akka.stream.testkit.scaladsl.{ TestSink, TestSource }
+import akka.util.{ ByteString, OptionVal }
+
+/**
+ * low-level tests testing Http2ServerDemux in isolation
```

Comment:
Can we put an end-to-end test into `H2cUpgradeSpec`, instead?

Reply:
Perhaps both? It might be helpful to document how `Http2ServerDemuxSpec` was supposed to behave in this low-level test